From 33a35a699928318032f62eab0e39e263c604ad9e Mon Sep 17 00:00:00 2001
From: Pino Toscano <pino@kde.org>
Date: Sat, 22 Oct 2022 13:51:17 +0200
Subject: [PATCH] hashunittest: run sha384longtest only for providers that
 support sha384

Fixes commit bdb0e9d37d78ec8e01d4008dba5243afe9029346 that accidentally
left the wrong logic when refactoring the file, and makes the check as
done in the other longtest tests.
---
 unittest/hashunittest/hashunittest.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/unittest/hashunittest/hashunittest.cpp b/unittest/hashunittest/hashunittest.cpp
index c2c25a0e..b208c94f 100644
--- a/unittest/hashunittest/hashunittest.cpp
+++ b/unittest/hashunittest/hashunittest.cpp
@@ -581,7 +581,7 @@ void HashUnitTest::sha384longtest()
     fillerString.fill('a', 1000);
 
     foreach (QString provider, providersToTest) {
-        if (!QCA::isSupported("sha384", provider)) {
+        if (QCA::isSupported("sha384", provider)) {
             // QTime t;
             // t.start();
             QCA::Hash shaHash(QStringLiteral("sha384"), provider);
-- 
2.38.0

